src/shared/parse-helpers.c

```diff
@@ -0,0 +1,23 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include "parse-helpers.h"
+#include "signal-util.h"
+#include "string-table.h"
+#include "string-util.h"
+
+int parse_arg_signal(const char *s, int *ret) {
```

Comment:
hmm, with this we now have 3 functions for the purpose of cmdline parsing. (the others are json_parse_cmdline_parameter_and_warn() and parse_path_argument_and_warn()

And they all have slightly different semantics and naming. Any chance this can be unified, in particular if you introduce a new source file for this?

i think the source should indicate this is about cmdline argument parsing btw, otherwise this is not easily distinguished from parse-util.h, which we already have. maybe: parse-cmdline-args.[ch]?